openpype/scripts/otio_burnin.py

```diff
@@ -589,59 +661,50 @@ def burnins_from_data(
             value = value.replace(SOURCE_TIMECODE_KEY, MISSING_KEY_VALUE)
 
-        # Convert lists.
-        cmd = ""
-        text = None
-        keys = [i[1] for i in Formatter().parse(value) if i[1] is not None]
-        list_to_convert = []
-
-        # Warn about nested dictionary support for lists. Ei. we dont support
-        # it.
-        if "[" in "".join(keys):
-            print(
-                "We dont support converting nested dictionaries to lists,"
-                " so skipping {}".format(value)
-            )
-        else:
-            for key in keys:
-                data_value = data[key]
-
-                # Multiple lists are not supported.
-                if isinstance(data_value, list) and list_to_convert:
-                    raise ValueError(
-                        "Found multiple lists to convert, which is not "
-                        "supported: {}".format(value)
-                    )
-
-                if isinstance(data_value, list):
-                    print("Found list to convert: {}".format(data_value))
-                    for v in data_value:
-                        data[key] = v
-                        list_to_convert.append(value.format(**data))
-
-        if list_to_convert:
-            value = list_to_convert[0]
-            path = convert_list_to_command(
-                list_to_convert, data["fps"], label=align
-            )
-            cmd = "sendcmd=f='{}'".format(path)
-            cmd = cmd.replace("\\", "/")
-            cmd = cmd.replace(":", "\\:")
-            clean_up_paths.append(path)
-
         # Failsafe for missing keys.
-        key_pattern = re.compile(r"(\{.*?[^{0]*\})")
         missing_keys = []
-        for group in key_pattern.findall(value):
+        listed_keys = {}
+        fill_values = {}
+        for item in Formatter().parse(value):
+            _, field_name, format_spec, conversion = item
+            if not field_name:
+                continue
+            keys = [key.rstrip("]") for key in field_name.split("[")]
+            conversion = "!{}".format(conversion) if conversion else ""
+            format_spec = ":{}".format(format_spec) if format_spec else ""
+            # Calculate original full key for replacement
+            orig_key = "{{{}{}{}}}".format(
+                field_name, conversion, format_spec)
+
+            key_value = data
             try:
-                group.format(**data)
-            except (TypeError, KeyError):
-                missing_keys.append(group)
+                for key in keys:
+                    key_value = key_value[key]
+
+                if isinstance(key_value, list):
+                    listed_keys[orig_key] = {
+                        "values": key_value,
+                        "keys": keys}
+                else:
+                    fill_values[orig_key] = orig_key.format(**data)
+            except (KeyError, TypeError):
+                missing_keys.append(orig_key)
+                continue
```

Comment:
Readability wise I think this is better put in a function separate from the class with a clear docstring:
```python

def format_value_array(value, data):
    """Format string with key-values in data.
    
    Data may contain lists for values so that each entry gets
    formatted on its own with resulting values.
    
    Arguments:
        value (str): String containing formatting values
        data (dict): Key-values used for formatting.
    
    Returns:
        tuple: listed_keys, fill_values, missing_keys
    
    """
    missing_keys = []
    listed_keys = {}
    fill_values = {}
    for item in Formatter().parse(value):
        _, field_name, format_spec, conversion = item
        if not field_name:
            continue
        keys = [key.rstrip("]") for key in field_name.split("[")]
        conversion = "!{}".format(conversion) if conversion else ""
        format_spec = ":{}".format(format_spec) if format_spec else ""
        # Calculate original full key for replacement
        orig_key = "{{{}{}{}}}".format(
            field_name, conversion, format_spec)

        key_value = data
        try:
            for key in keys:
                key_value = key_value[key]

            if isinstance(key_value, list):
                listed_keys[orig_key] = {
                    "values": key_value,
                    "keys": keys}
            else:
                fill_values[orig_key] = orig_key.format(**data)
        except (KeyError, TypeError):
            missing_keys.append(orig_key)
            continue
            
    return listed_keys, fill_values, missing_keys
    
```